app/imports/components/fields/UploadField.jsx

```diff
@@ -40,9 +40,19 @@ class FieldUpload extends Component {
     }.bind(this));
   }
 
+  handleClick() {
+    this.refs.fileInput.dispatchEvent(new MouseEvent('click', {
+      'view': window,
+      'bubbles': true,
+      'cancelable': true
+    }));
+  }
+
   render() {
+    let displayTrigger = (!this.state.uploading || _.isEmpty(this.props.fileUrl)) ? 'inline-block' : 'none';
```

Comment:
I think this is a bad pattern. It would be better for the whole `<section>` of the `upload-container` to have a class of `uploading` or similar while it is uploading and to use that class to show/hide the file upload dialog button. Then in the future we could use the same class to show other UI elements related to upload